src/func/externalServiceOauth.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2022  Ansh Goyal
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import type {ExternalServiceTokenT} from './types';
+
+/**
+ * Fetches the Oauth token with related data
+ * @param {number} editorId - Editor's id.
+ * @param {string} service - Name of the service.
+ * @param {object} orm - the BookBrainz ORM, initialized during app setup
+ * @returns {Promise} A Promise that resolves to the oauth token in JSON format
+ */
+export async function getOauthToken(
+	editorId: number,
+	service: string,
+	orm: Record<string, any>
+): Promise<ExternalServiceTokenT> {
+	const rawSql =
+		`SELECT 
+			  editor_id,
+			  service,
+			  access_token,
+			  refresh_token,
+			  token_expires::text,
+              scopes
+		 FROM external_service_oauth 
```

Comment:
I merged the PR too quickly ! 😱 

I get an error when I run the tests locally: all these queries should reference `bookbrainz.external_service_oauth` rather than just `external_service_oauth`.
I get errors like so: "relation "external_service_oauth" does not exist".

Would you mind opening a new PR with the fixes?

Reply:
Sure! I'll open a PR